packages/plugin-ext/src/main/browser/languages-main.ts

```diff
@@ -797,16 +796,14 @@ function reviveOnEnterRules(onEnterRules?: SerializedOnEnterRule[]): monaco.lang
     return onEnterRules.map(reviveOnEnterRule);
 }
 
-export function reviveWorkspaceEditDto(data: WorkspaceEditDto): monaco.languages.WorkspaceEdit {
-    if (data && data.edits) {
-        for (const edit of data.edits) {
-            if (typeof (<ResourceTextEditDto>edit).resource === 'object') {
-                (<ResourceTextEditDto>edit).resource = URI.revive((<ResourceTextEditDto>edit).resource);
+export function toMonacoWorkspaceEdit(data: WorkspaceEditDto): monaco.languages.WorkspaceEdit {
```

Comment:
Can we guarantee that data and data.edits are not undefined? It looks like the previous code had guards to check for it

Reply:
thanks for testing @JPinkney and @benoitf!
indeed this was not the case with the happy path.

given this shape:
```
export interface WorkspaceEditDto {
    edits: (ResourceFileEditDto | ResourceTextEditDto)[];
    rejectReason?: string;
}
```
I adjusted the result of `provideRenameEdits` in case of an error to not return `undefined!` for `edits`. 

actually, I don't understand why the adapter [isn't rejecting in this case](https://github.com/eclipse-theia/theia/blob/4beeedc2410540e2a6e236a25e6e097bc2cffa8b/packages/plugin-ext/src/plugin/languages/rename.ts#L57) but returning with something to be handled down the road.  well, I couldn't find any use of `rejectReason` either. what's the concept behind this?

Reply:
> Can we guarantee that data and data.edits are not undefined?

@JPinkney as long as we're are creating these objects, the answer is "yes".

Reply:
@JPinkney I had to update again. Please retest.